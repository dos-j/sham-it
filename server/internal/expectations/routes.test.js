const registerRoutes = require("./routes");

describe("unit: internal/expectations/routes", () => {
  let routes;
  let utils;

  beforeEach(() => {
    utils = {
      internalRoute(...args) {
        return args;
      },
      requestStore: []
    };

    routes = registerRoutes(utils);
  });

  test("it should return all of the expectation routes", () => {
    expect(routes).toMatchSnapshot();
  });
});
