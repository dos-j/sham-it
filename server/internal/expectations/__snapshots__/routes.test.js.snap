// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit: internal/expectations/routes it should return all of the expectation routes 1`] = `
Array [
  Array [
    "POST",
    /\\^\\\\/\\\\\\$hasbeencalled\\$/,
    [Function],
  ],
  Array [
    "POST",
    /\\^\\\\/\\\\\\$not\\\\/hasbeencalled\\$/,
    [Function],
  ],
  Array [
    "POST",
    /\\^\\\\/\\\\\\$hasbeencalledtimes\\$/,
    [Function],
  ],
  Array [
    "POST",
    /\\^\\\\/\\\\\\$not\\\\/hasbeencalledtimes\\$/,
    [Function],
  ],
  Array [
    "POST",
    /\\^\\\\/\\\\\\$hasbeencalledwith\\$/,
    [Function],
  ],
  Array [
    "POST",
    /\\^\\\\/\\\\\\$not\\\\/hasbeencalledwith\\$/,
    [Function],
  ],
  Array [
    "POST",
    /\\^\\\\/\\\\\\$hasbeenlastcalledwith\\$/,
    [Function],
  ],
  Array [
    "POST",
    /\\^\\\\/\\\\\\$not\\\\/hasbeenlastcalledwith\\$/,
    [Function],
  ],
]
`;
